ksiren-gson-adapter/src/main/java/com/brightspace/ksiren/gson_adapter/KSirenGsonAdapter.kt

```diff
@@ -44,15 +44,15 @@ class KSirenGsonAdapter(val gsonReader: JsonReader) : KSirenJsonReader {
 		return gsonReader.nextName()
 	}
 
-	override fun nextString(): String {
-		try {
-			return gsonReader.nextString()
-		} catch( e: Exception) {
-			throw KSirenException.ParseException(e.message?:"Reading string failed")
-		}
+	override fun nextStringImpl(): String {
```

Comment:
Do we still need this method even with the `nextString()` one being able to handle nulls?

Reply:
We do, because we still need to allow the adapters to define what reading a string means for them. The concrete `readString()` function in the abstract class is really only telling the implementation what to do, and not how to do it.

Reply:
The implementation of the method should be enough for each adapter. The interface already reflects it's a nullable string so we can deal with it in any way we want. It seems `nextStringImpl()` is just taking care of a specific scenario that doesn't need to be separated from the others.

Reply:
We still need a way for `nextString()` to call either "gson.nextString()` in the case of the GsonAdapter or `moshi.nextString()` in the case of the moshi adapter, we don't have the ability to make those calls from the interface.